src/components/TheCanvas.vue

```diff
@@ -63,69 +68,47 @@ export default class TheCanvas extends Vue {
           }
         }
       }, true)
+
       this.socket.onmessage = (data: MessageEvent) => {
         try {
-          const receivedData = JSON.parse(data.data).payload
-          if (receivedData.type && receivedData.data.e) {
-            this.enableTool(receivedData.type)
-            this.throttleToolRender(receivedData.data.e, true)
+          const canvasElement: CanvasElement = JSON.parse(data.data).payload
+          if (canvasElement.immediate) {
+            // render it on the canvas
+            const foundTool = this.tools.find((tool: Tool) => tool.name === canvasElement.tool.name)
+            if (foundTool && foundTool.renderCanvas && canvasElement.jti !== 'SAM') {
+              foundTool.renderCanvas(canvasElement, this.layerNode)
+            }
+          } else {
+            // send it to the store and then render it on canvas
+            this.addCanvasElement(canvasElement)
           }
-        } catch (err) {
-        }
+        } catch (err) { }
```

Comment:
Comment, why we are not doing anything to catched error.

Reply:
we need to talk about the error handling for the app. Maybe a logging suite like Winston might do the job?

Reply:
Sentry?

Reply:
Let's open a discussion in discord about this :)